pwa-devdocs/src/tutorials/create-taglist-component/index.md

```diff
@@ -0,0 +1,454 @@
++    const url = `/${url_path}${categoryUrlSuffix}`;
+ 
+-    // Return the tag string inside a list item element
+-    return (<li>{value}</li>);
++    // Return the tag as a Link component wrapped around a Button
++    return (
++        <Link to={url}>
++            <Button priority="low" type="button">
++                {name}
++            </Button>
++        </Link>
++    );
+ }
+ 
+ // Define the props this component accepts
+ Tag.propTypes = {
+-    value: string
++    value: shape({
++       name: string,
++       url_path: string
++    })
+ }
+ 
+ // Make this function the default exported module for this file
+ export default Tag;
+```
+
+This update imports existing PWA Studio components and uses them to compose a tag button that acts as a link.
+
+### Update TagList component
+
+Since the Tag component no longer returns the tag inside a list item element, the TagList component needs to be updated to pass in the keyword object and replace the `ul` container with a `div` container:
+
+```diff
+ const tagList = categories.map(keyword => {
+-    return <Tag key={keyword.name} value={keyword.name} />;
++    return <Tag key={keyword.name} value={keyword} />;
+ });
+-return(<ul>{tagList}</ul>)
++return(<div>{tagList}</div>)
+```
+
+Now, when you open the product detail page, you see the tags listed horizontally and clicking on one takes you to that category's page.
+
+![Unstyled list of tag buttons](images/unstyled-list.png)
+
+## Step 8. Style the components
+
+With the click functionality in place, the last step is to add custom styles to make the tags stand out.
+
+### Define the TagList style
+
+A product can fall into multiple categories, so
+the TagList style must accomodate for any number of categories.
+
+#### `tagList.css`
+
+Open the `tagList.css` file and add the following content:
+
+```css
+.root {
+    display: flex;
+    flex-wrap: wrap;
+}
+```
+
+This entry defines the root container of a TagList component as a [flexbox][] that wraps.
+This prevents the Tags from overflowing off the side of the page.
+
+#### `tagList.js`
+
+Open the `tagList.js` file and make the following changes to the component.
+
+**Change 1:** Import the `mergeClasses()` function and the `tagList.css` file:
+
+```diff
+ import { arrayOf, shape, string } from 'prop-types';
+ 
++import { mergeClasses } from '../../classify';
++import defaultClasses from './tagList.css';
+ 
+ import Tag from './tag';
+```
+
+**Change 2:** Use the `mergeClasses()` function to merge custom classes passed into the component:
+
+```diff
+ const { categories } = props;
+ 
++const classes = mergeClasses(defaultClasses, props.classes);
+```
+
+{: .bs-callout .bs-callout-info}
+The `mergeClasses()` function is used for merging a component's default classes with custom class overrides.
+
+**Change 3:** Set the `className` property of the `div` container to the `root` class:
+
+```diff
+-return(<div>{tagList}</div>)
++return(<div className={classes.root}>{tagList}</div>)
+```
+
```

Comment:
I don't know if you care about updating `Tag.propTypes` here but you could add `classes` to it.